pkg/k8s/apis/cilium.io/v2alpha1/bgp_node_types.go

```diff
@@ -0,0 +1,183 @@
+	LocalAddress *string `json:"localAddress,omitempty"`
+
+	// PeerConfigRef is a reference to a peer configuration resource.
+	//
+	// +kubebuilder:validation:Optional
+	PeerConfigRef *PeerConfigReference `json:"peerConfigRef,omitempty"`
+}
+
+// CiliumBGPNodeStatus is the status of the CiliumBGPNodeConfig.
+type CiliumBGPNodeStatus struct {
+	// BGPInstances is the status of the BGP instances on the node.
+	//
+	// +listType=map
+	// +listMapKey=localASN
+	BGPInstances []CiliumBGPNodeInstanceStatus `json:"bgpInstances"`
+}
+
+type CiliumBGPNodeInstanceStatus struct {
+	// LocalASN is the ASN of this BGP instance.
+	//
+	// +kubebuilder:validation:Required
+	LocalASN int64 `json:"localASN"`
+
+	// PeerStatuses is the state of the BGP peers for this BGP instance.
+	//
+	// +listType=map
+	// +listMapKey=peerAddress
+	PeerStatuses []CiliumBGPNodePeerStatus `json:"peers"`
+}
+
+// CiliumBGPNodePeerStatus is the status of a BGP peer.
+type CiliumBGPNodePeerStatus struct {
+	// PeerAddress is the IP address of the neighbor.
+	//
+	// +kubebuilder:validation:Required
+	PeerAddress string `json:"peerAddress"`
+
+	// PeerASN is the ASN of the neighbor.
+	//
+	// +kubebuilder:validation:Required
+	PeerASN int64 `json:"peerASN"`
+
+	// PeeringState is last known state of the peering session.
+	PeeringState string `json:"peeringState,omitempty"`
+
+	// Timers is the state of the negotiated BGP timers for this peer.
+	Timers CiliumBGPTimersState `json:"timers,omitempty"`
+
+	// Uptime is the time since the last peering session was established.
+	Uptime string `json:"uptime,omitempty"`
+
+	// RoutesReceived is the number of routes received from this peer.
+	RoutesReceived int32 `json:"routesReceived,omitempty"`
+
+	// RoutesAdvertised is the number of routes advertised to this peer.
+	RoutesAdvertised int32 `json:"routesAdvertised,omitempty"`
```

Comment:
These fields should be pointers and marked with `// +kubebuilder:validation:Optional` if they will be omitted from JSON serialization.